torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -513,6 +513,17 @@ class ShapePropagator {
         }
         return;
       }
+      case prim::unchecked_unwrap_optional: {
+        // If we have None as input, we need to leave the output type alone
+	// note that we cannot have None as input in execution,
+	// but we want to keep the graph consistent during the analysis
+        if (auto ot = node->input()->type()->cast<OptionalType>()) {
+          node->output()->setType(ot->getElementType());
+        } else if (!node->input()->type()->isSubtypeOf(NoneType::get())) {
+          node->output()->setType(node->input()->type());
+        }
+        return;
```

Comment:
When do we emit `unchecked_unwrap_optional`s though? It seems that if they're unchecked then they should never see `NoneType`s as inputs, or they would silently result in UB at runtime? I don't understand what does it mean to "keep the graph consistent". This simply looks wrong

Reply:
Well, we know that the if branch containing `unchecked_unwrap_optional` will never be executed. However, the branch is still there during the various JIT passes.

Reply:
They don't see None anymore.